src/client/datascience/jupyter/kernels/types.ts

```diff
@@ -53,7 +53,19 @@ export type KernelSpecConnectionMetadata = Readonly<{
      * This interpreter could also be the interpreter associated with the kernel spec that we are supposed to start.
      */
     interpreter?: PythonEnvironment;
-    kind: 'startUsingKernelSpec';
+    kind: 'startUsingLocalKernelSpec';
+    id: string;
+}>;
+/**
+ * Connection metadata for Remote Kernels started using kernelspec (JSON).
+ * This could be a raw kernel (spec might have path to executable for .NET or the like).
+ * If the executable is not defined in kernelspec json, & it is a Python kernel, then we'll use the provided python interpreter.
+ */
+export type RemoteKernelSpecConnectionMetadata = Readonly<{
+    kernelModel?: undefined;
+    interpreter?: undefined;
```

Comment:
Remote kernels shouldn't have interpreters, right? There's no interpreter on the remote machine.

Reply:
Yes, but i've had to add this property because we try to check the property `interpreter` in the unified type.
If i don't add this then code won't compile after all, not all kernelspecconnection types will have `interpertetr`.
Hence I've added a dummy propety that will always be either undefied or non-existent.

Reply:
Fixed